src/Sulu/Bundle/ContentBundle/Preview/PreviewMessageHandler.php

```diff
@@ -0,0 +1,302 @@
+        }
+
+        return $result;
+    }
+
+    /**
+     * Reconnect to mysql
+     */
+    private function reconnect()
+    {
+        /** @var EntityManagerInterface $entityManager */
+        $entityManager = $this->registry->getManager();
+        /** @var Connection $connection */
+        $connection = $entityManager->getConnection();
+
+        try {
+            $connection->executeQuery('SELECT 1;');
+        } catch (DBALException $exc) {
+            $this->logger->warning('Mysql reconnect');
+            $connection->close();
+            $connection->connect();
+        }
+    }
+
+    /**
+     * Start preview session
+     * @param ConnectionInterface $conn
+     * @param MessageHandlerContext $context
+     * @param array $msg
+     * @return array
+     * @throws MissingParameterException
+     */
+    private function start(ConnectionInterface $conn, MessageHandlerContext $context, $msg)
+    {
+        // init session
+
+        // locale
+        if (!array_key_exists('locale', $msg)) {
+            throw new MissingParameterException('locale');
+        }
+        $locale = $msg['locale'];
+        $context->set('locale', $locale);
+
+        // webspace key
+        if (!array_key_exists('webspaceKey', $msg)) {
+            throw new MissingParameterException('webspaceKey');
+        }
+        $webspaceKey = $msg['webspaceKey'];
+        $context->set('webspaceKey', $webspaceKey);
+
+        // user id
+        if (!array_key_exists('user', $msg)) {
+            throw new MissingParameterException('user');
+        }
+        $user = $msg['user'];
+        $context->set('user', $user);
+
+        // content uuid
+        if (!array_key_exists('content', $msg)) {
+            throw new MissingParameterException('content');
+        }
+        $contentUuid = $msg['content'];
+        // filter index page
+        if ($contentUuid === 'index') {
+            $startPage = $this->contentMapper->loadStartPage($webspaceKey, $locale);
+            $contentUuid = $startPage->getUuid();
+        }
+        $context->set('content', $contentUuid);
+
+        // init message vars
+        $template = array_key_exists('template', $msg) ? $msg['template'] : null;
+        $data = array_key_exists('data', $msg) ? $msg['data'] : null;
+
+        // start preview
+        $this->preview->start($user, $contentUuid, $webspaceKey, $locale, $data, $template);
+
+        return array(
+            'command' => 'start',
+            'content' => $contentUuid,
+            'msg' => 'OK'
+        );
+    }
+
+    /**
+     * Stop preview session
+     * @param ConnectionInterface $from
+     * @param MessageHandlerContext $context
+     * @return array
+     * @throws ContextParametersNotFoundException
+     */
+    private function stop(ConnectionInterface $from, MessageHandlerContext $context)
+    {
+        // check context parameters
+        if (!$context->has('user')) {
+            throw new ContextParametersNotFoundException();
```

Comment:
Is it possible that you use this exception quite often? Because its description says that it is thrown when the preview is not running.